src/olympia/ratings/views.py

```diff
@@ -392,8 +392,34 @@ def get_paginated_response(self, data):
             if show_grouped_ratings and self.get_addon_object():
                 response.data['grouped_ratings'] = dict(GroupedRating.get(
                     self.addon_object.id))
+        if 'addon' in self.request.GET and is_gate_active(
+                self.request, 'ratings-can_reply'):
+            response.data['can_reply'] = (
+                bool(self.request.user.is_authenticated) and
+                self.check_can_reply_permission_for_ratings_list())
         return response
 
+    def check_can_reply_permission_for_ratings_list(self):
+        """Check whether or not the current request contains an user that can
+        reply to ratings we're about to return.
+
+        Used to populate the `can_reply` property in ratings list, when an
+        addon is passed."""
+        # Clone the current viewset, but change permission_classes.
+        viewset = self.__class__(**self.__dict__)
+        viewset.permission_classes = self.reply_permission_classes
+
+        # Create a fake rating with the addon object attached, to be passed to
+        # check_object_permissions().
+        dummy_rating = Rating(addon=self.get_addon_object())
```

Comment:
I'm not sure how we can get around this, but the `is_moderator` clause won't ever be true because it needs the actual `Rating` and we're just returning `can_reply` at the top level.  We should mention this somewhere.

Reply:
Irrelevant because it's not `CanDeleteRatingPermission` we're using

Reply:
(I did think about also supporting `can_delete` but we're moving away from moderators/admin special actions directly on the frontend so I thought it wasn't worth it)